src/otaclient_common/linux.py

```diff
@@ -252,3 +258,33 @@ def copyfile_nocache(src: StrOrPath, dst: StrOrPath) -> None:
         finally:
             os.posix_fadvise(src_fd, 0, 0, os.POSIX_FADV_DONTNEED)
             os.posix_fadvise(dst_fd, 0, 0, os.POSIX_FADV_DONTNEED)
+
+
+#
+# ------ fstrim with timeout ------ #
+#
+
+
+def fstrim_at_thread(
+    target_mountpoint: Path, *, waited: bool = True, timeout: float = 120
+) -> bool:  # pragma: no cover
+    """Dispatch a sub-thread for running fstrim with timeout.
+
+    fstrim is safe to be interrupted at any time.
+
+    Returns:
+        True if fstrim finished without reaching timeout.
+    """
+
+    def _thread_entry():
+        _cmd = ["fstrim", "-v", str(target_mountpoint)]
+        subprocess.run(_cmd, timeout=timeout)
```

Comment:
How about to use `subprocess_run_wrapper` instead?

Reply:
Here we don't need the output of the command execution, neither need to check the execution result. We only need to dispatch the execution at the background thread without any inspection, so plain `subprocess.run` is totally enough.

Reply:
( Also, the target_mountpoint is visible within the dynamic otaclient mount namespace, and the `fstrim` presents within the otaclient squashfs image(installed in the `ubuntu` docker image by default), so no need to do chroot for the command execution.